web/src/components/workflow-graph/node-base.tsx

```diff
@@ -62,6 +63,30 @@ export default function NodeBase({
 				</Flex>
 			</Card>
 
+			<NodeToolbar
+				isVisible={hasMissingSecret(nodeId)}
+				position={Position.Top}
+				offset={8}
+				align="end"
+			>
+				<Tooltip content="Missing secret">
+					<Flex
+						style={{
+							transform: "translate(10px, 24px)",
+						}}
+					>
+						<Flex className="absolute inset-0 bg-[var(--red-9)] rounded-full " />
```

Comment:
can you remove the whitespace at the end?

```suggestion
						<Flex className="absolute inset-0 bg-[var(--red-9)] rounded-full" />
```